graphql/util.go

```diff
@@ -1080,6 +1080,16 @@ func userHasDistroPermission(u *user.DBUser, distroId string, requiredLevel int)
 	return u.HasPermission(opts)
 }
 
+func userHasHostPermission(u *user.DBUser, distroId string, requiredLevel int) bool {
+	opts := gimlet.PermissionOpts{
```

Comment:
I think it could make sense to check if the user is the owner as the first condition in this function. If a user owns the host, then they should be able to do whatever they want

That might get rid of the need for `SPAWN_EDIT`

Reply:
This change would make the API/code simpler but would make the permissions less strict than what EVG currently has in place (check `ModifyHostsWithPermissions`). I think in general, basing the permission check off of startedBy is not an ideal pattern since it exists outside of the permission framework and would like to keep it siloed to spawn host since that's what has been in place (check `CanUpdateSpawnHost`).